### PR TITLE
[KYUUBI #2704] verify TPC-DS query output

### LINKED ISSUE #2704: [Subtask] Kyuubi Spark TPC-DS Connector - Verify TPC-DS query output
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

Verify TPC-DS query output, e.g. `sum(hash(*))`, result schema

### Parent issue

https://github.com/apache/incubator-kyuubi/issues/2538

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2704 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala [hunk 1]
```diff
@@ -17,36 +17,100 @@
...
         try {
-          spark.sql(sql).collect()
+          val result = spark.sql(sql).collect()
+          val schema = spark.sql(sql).schema
+          val schemaDDL = schema.toDDL + "\n"
+          spark.createDataFrame(result.toList.asJava, schema).createTempView(s"$name$viewSuffix")
+          val sumHashResult =
+            spark.sql(s"select sum(hash(*)) from $name$viewSuffix").collect().head.get(0) + "\n"
+
+          // scalastyle:off println
```
Comment (jiaoqingbo) [inline extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala:90]: The sumHashResult I calculated locally is different from the remote CI's, wait for the CI to execute to get the results and delete these

**Comment by jiaoqingbo** [comment]
It seems that a license header needs to be added to the new file

**Review by yaooqinn — APPROVED** [review]
Not related to this PR, shall we merge tpcdh/tpcds golden files into one hash and one schema files as each of them has one one row

**Comment by jiaoqingbo** [comment]
> Not related to this PR, shall we merge tpcdh/tpcds golden files into one hash and one schema files as each of them has one one row

I will change TPCH to be the same as TPCDS in a follow up pr of #2805

**Comment by pan3793** [comment]
Seems unnecessary to move to sub dir, it's good that queries and outputs share the same folder

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala (hunk 1 above)
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala]: Do you find the root cause? Are you using ARM CPU like Apple Silicon in local?

Comment (jiaoqingbo) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala]: My local is a windows development environment, is it related to jdk? CI is openJdk, I am not local

**Comment by jiaoqingbo** [comment]
> outputs

ok I will change it

**Comment by jiaoqingbo** [comment]
> Seems unnecessary to move to sub dir, it's good that queries and outputs share the same folder

I will change it

File: extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala (hunk 1 above)
Comment (pan3793) [reply extensions/spark/kyuubi-spark-connector-tpcds/src/test/scala/org/apache/kyuubi/spark/connector/tpcds/TPCDSQuerySuite.scala]: Technically, the result is affectted by the algorithm, the seed, and the memory bits of the value. I'm not sure ARM, but the x86 based JDK should has a deterministic memory layout

**Comment by jiaoqingbo** [comment]
> Not related to this PR, shall we merge tpcdh/tpcds golden files into one hash and one schema files as each of them has one one row

I misunderstood what you said before, it seems more convenient to place them separately for comparison

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

**Pull request closed without merging**